vmware-event-router/internal/processor/aws_event_bridge.go

```diff
@@ -221,23 +223,32 @@ func (awsEventBridge *awsEventBridgeProcessor) createPutEventsInput(baseEvent []
 		cloudEvent := events.NewCloudEvent(event, eventInfo, awsEventBridge.source)
 		jsonBytes, err := json.Marshal(cloudEvent)
 		if err != nil {
-			return eventbridge.PutEventsInput{}, errors.Wrapf(err, "could not marshal cloud event for vSphere event %d from source %s", event.GetEvent().Key, awsEventBridge.source)
+			return []eventbridge.PutEventsInput{}, errors.Wrapf(err, "could not marshal cloud event for vSphere event %d from source %s", event.GetEvent().Key, awsEventBridge.source)
 		}
-
 		jsonString := string(jsonBytes)
 		entry := eventbridge.PutEventsRequestEntry{
 			Detail:       aws.String(jsonString),
 			EventBusName: aws.String(awsEventBridge.patternMap[eventInfo.Name]),
 			Source:       aws.String(cloudEvent.Source),
 			DetailType:   aws.String(cloudEvent.Subject),
 		}
-		input.Entries = append(input.Entries, &entry)
+		tmpInput.Entries = append(tmpInput.Entries, &entry)
 
 		// update metrics
 		awsEventBridge.stats.Invocations[eventInfo.Name]++
+
+		if idx%batchSize == 0 && idx != 0 {
```

Comment:
Please help me to understand the `&& idx != 0`

Reply:
Yeah so this logic here is used only if the batches are more than 10. In case for example the input entries are 0-9 we want a single batch which is created on line 247. This is just that 0 % 0 == 0 and we will have 2 batches instead of the desired one. So if we have 3 input entries and remove this we will have 1 batch with 1 input entry and then a second batch with 3 input entries when we only want 1 batch with 3 input entries.